submissions/Semka094/frogger/app.js

```diff
@@ -0,0 +1,127 @@
+const gameLevel = document.createElement('div');
+gameLevel.textContent = 'Your level: 0';
+
+document.body.append(gameLevel);
+
+const xStep = 100;
+const yStep = 85;
+
+const coordinates = {
+  startEnemyXPosition: -90,
+  winPosition: -45,
+  defaultPlayerXPosition: 200,
+  defaultPlayerYPosition: 380,
+  maxYDownPosition: 380,
+  maxXRightPosition: 400,
+  maxXLeftPosition: 0,
+  enemyPositionYDiff: 20,
+};
+
+class Enemy {
+  sprite = 'images/enemy-bug.png';
+
+  constructor({speed, x, y}) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+  }
+
+  update(dt) {
+    this.x += this.speed * dt;
+    
+    // Cross the boundaries of the board - move to start position (endless movement) 
+    if (this.x >= ctx.canvas.width) {
+      this.x = coordinates.startEnemyXPosition;
+    }
+    
+    // Check end game; Check loose only for the same Y
+    if (player.y === (this.y - coordinates.enemyPositionYDiff)) {
+      if (
+        player.x < this.x && (this.x - player.x) <= 50 || // Player is behind the bug
+        player.x > this.x && (player.x - this.x) <= 50    // Player is in front of the bug
+      ) {
+        alert('Game over!');
+        player.resetGame();
+      }
```

Comment:
Items 4.iii and 4.iv of the requirements.

Reply:
Do you mean these [requirements](https://docs.google.com/document/d/1v01aScPjSWCCWQLIpFqvg3-vXLH2e8_SZQKC8jNO0Dc/pub) ?

>Recall that the player cannot move off screen (so you will need to check for that and handle appropriately).

>If the player reaches the water the game should be reset by moving the player back to the initial location (you can write a separate reset Player method to handle that).

If yes, it is implemented.

Reply:
I mean these https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger